src/hooks/useForm/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useMemo } from 'react';
+
+// Interfaces START
+
+export interface Validator {
+  (value: string): string;
+}
+
+interface FormInput {
+  value: string;
+  validators: Validator[];
+}
+
+interface ParameterValueFunction<T> {
+  (formInput: FormInput, key: string): T;
+}
+
+interface Dictionary<T> {
+  [key: string]: T;
+}
+
+// Interfaces END
+
+// Hook START
+const useForm = (fieldsSettings: Dictionary<FormInput>) => {
+  const convertArrayToObject = (
+    dataSource: Dictionary<FormInput>,
+    createValue: ParameterValueFunction<any>
+  ) =>
+    Object.entries(dataSource).reduce(
```

Comment:
I have the feeling that we are using reduce just for its own iteration functionality. Here you start with an object of n properties (n = number of [properties/values] ). You convert it into an array of the same length n (Object.entries(...)). Finally you map that array and construct an object from its [key,value] pairs.
I don't see why dont we just use a loop instead:

```
let obj = {};
for (let [key, value] of Object.entries(fieldsSettings)) {
   object[key] = createValue(value, key)
}
return object;
```
 A little comment: before fixing this comment please read the comment related to some data structures.

Reply:
I loved this comment, you are right, when the input data structure changes I forgot to look at this method in depth so I agree with you to make this change.